src/fu-keyring.c

```diff
@@ -198,222 +70,63 @@ fu_keyring_add_public_keys (FuKeyring *keyring, const gchar *dirname, GError **e
 		if (filename == NULL)
 			break;
 		path_tmp = g_build_filename (dirname, filename, NULL);
-		if (!fu_keyring_add_public_key (keyring, path_tmp, error))
-			return FALSE;
+		for (guint i = 0; i < priv->cryptos->len; i++) {
+			FuCrypto *crypto = g_ptr_array_index (priv->cryptos, i);
+			g_debug ("adding public key %s to crypto provider %s",
+				 path_tmp, fu_crypto_get_name (crypto));
+			if (!fu_crypto_add_public_key (crypto, path_tmp, error))
```

Comment:
I'm not terribly happy about putting both the GPG public key and the PKCS certifcate in the same directory; I did wonder if the PCKS7 certificate should go in something like `/etc/pki/tls/certs/` instead. Perhaps we just install the gpg key to `/etc/pki/fwupd/gpg` and search for PKCS keys in `/etc/pki/fwupd/pkcs7` or something -- at the moment it feels like a hack returning TRUE if the key type isn't valid for the crypto provider.